### ISSUE TITLE
WireGuard support in VPN profiles 

### ISSUE THREAD

**Issue by ghost** [GrapheneOS/os_issue_tracker#2740]
The current version of AOSP only supports IKEv2/IPSec protocol when users try to add a VPN profile manually. Will it be possible to add WireGuard here; so that users can connect to VPN servers without installing any kind of VPN apps?

**Comment by thestinger** [comment]
Yes, it will be possible for WireGuard to be added there for the Pixel 6 and later where kernel support is available.

**Comment by thestinger** [comment]
We expect Android to implement it and don't really plan to do it ourselves.

**Comment by mpfaff** [comment]
Do you still expect Android to implement this in a reasonable time frame?

**Comment by MulverineX** [comment]
@thestinger I seriously doubt Android upstream is ever actually going to implement Wireguard support, its been many years without any actual signs of work. A GrapheneOS-specific implementation is probably needed here if support within a decade is desired.

**Comment by MulverineX** [comment]
Additionally, Tailscale (an open protocol with open source implementations used by many power users seeking to use GrapheneOS) support is necessary IMO. GrapheneOS is supposed to push for user sovereignty and security, but without magisk-tailscaled, a lot of self-hosted service alternatives are not easily, safely, & efficiently accessible.

Either this, or some sort of custom GrapheneOS-specific API that provides more reliable and efficient custom VPN capability to apps that implement it.

You should not need to root or replace your battery every 6 months to have an always-on Intranet.
